client/app/src/test/java/com/example/criengine/BookUnitTest.java

```diff
@@ -0,0 +1,165 @@
+package com.example.criengine;
+
+import com.example.criengine.Objects.Book;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.List;
+import static org.junit.Assert.*;
+
+/**
+ * Unit tests for the Book.java class.
+ * These tests test every method in the class.
+ */
+public class BookUnitTest {
+    Book mockBook;
+    ArrayList<String> newRequesterList;
+
+    @Before
+    public void init() {
+        newRequesterList = new ArrayList<>();
+        mockBook = new Book("ID 1", "New Owner", "New Title", "New Author", "New Description", "New ISBN", "New Status", "New Borrower", newRequesterList, "New Geolocation", "New Image");
+    }
+
+    @Test
+    public void testBookConstructorOne() {
+        Book mockBook = new Book();
+
+        assertNull(mockBook.getBookID());
+        assertNull(mockBook.getOwner());
+        assertNull(mockBook.getTitle());
+        assertNull(mockBook.getAuthor());
+        assertNull(mockBook.getDescription());
+        assertNull(mockBook.getIsbn());
+        assertNull(mockBook.getStatus());
+        assertNull(mockBook.getBorrower());
+        assertEquals(mockBook.getRequesters().size(), 0);
+        assertNull(mockBook.getGeolocation());
+        assertNull(mockBook.getImageURL());
+    }
+
+    @Test
+    public void testBookConstructorTwo() {
+        assertEquals(mockBook.getBookID(), "ID 1");
+        assertEquals(mockBook.getOwner(), "New Owner");
+        assertEquals(mockBook.getTitle(), "New Title");
+        assertEquals(mockBook.getAuthor(), "New Author");
+        assertEquals(mockBook.getDescription(), "New Description");
+        assertEquals(mockBook.getIsbn(), "New ISBN");
+        assertEquals(mockBook.getStatus(), "New Status");
+        assertEquals(mockBook.getBorrower(), "New Borrower");
+        assertEquals(mockBook.getRequesters().size(), 0);
+        assertEquals(mockBook.getGeolocation(), "New Geolocation");
+        assertEquals(mockBook.getImageURL(), "New Image");
+    }
+
+    @Test
+    public void testBookConstructorThree() {
+        List<String> newRequesterList = new ArrayList<>();
```

Comment:
Unneeded

Reply:
That is not part of the default JUnit provided in android studios.

Reply:
Alright then, sounds good